module/VuFind/src/VuFind/Marc/MarcReader.php

```diff
@@ -84,13 +85,19 @@ public function __construct($data)
     /**
      * Set MARC record data
      *
-     * @param string $data MARC record in MARCXML or ISO2709 format
+     * @param string|array $data MARC record in MARCXML or ISO2709 format, or an
+     * associative array with 'leader' and 'fields' in the internal format
      *
      * @throws Exception
      * @return void
      */
-    public function setData(string $data): void
+    public function setData($data): void
     {
+        if (is_array($data)) {
+            $this->leader = $data['leader'];
```

Comment:
Should we validate that the array comes in the expected format instead of assuming that leader and fields will be set?

Reply:
I added checks for the basic elements, but I don't think it's worth it checking the full structure.